Source/UI/PeersView.swift

```diff
@@ -153,6 +159,25 @@ class PeersView: UIView {
         label.attributedText = attributed
     }
     
+    
+    @objc func triggerSync(sender : UITapGestureRecognizer) {
+        self.connectionAnimation.searchAnimation()
+        
+        DispatchQueue.global(qos:  .userInitiated).async {
```

Comment:
We should not call Sync and Refresh directly, we need to use SyncOperation and RefreshOperation instead. Also, we need to encapsulate them inside IOS BackgroundTasks so that we are sure they finish if the user sends the app to the background. I think we can just call AppController.shared.pokeSync() here that address everything there. And for the long refresh, we have a similar pokeRefresh() function, we could add a RefreshLoad argument there so we can continue doing tiny refresh every 15 secs and do a long refresh here.

Reply:
Ok, fixed it so we're using AppController.shared.pokeSync()